boot/seal.go

```diff
@@ -95,27 +102,28 @@ func sealKeyToModeenv(key secboot.EncryptionKey, model *asserts.Model, modeenv *
 		bootloader.RoleRunMode:  bl.Name(),
 	}
 
-	// get model parameters from bootchains
-	modelParams, err := sealKeyModelParams(pbc, roleToBlName)
-	if err != nil {
-		return fmt.Errorf("cannot prepare for key sealing: %v", err)
-	}
 	// make sure relevant locations exist
 	for _, p := range []string{InitramfsEncryptionKeyDir, InstallHostFDEDataDir, InstallHostFDESaveDir} {
 		// XXX: should that be 0700 ?
 		if err := os.MkdirAll(p, 0755); err != nil {
 			return err
 		}
 	}
-	sealKeyParams := &secboot.SealKeyParams{
-		ModelParams:          modelParams,
-		KeyFile:              filepath.Join(InitramfsEncryptionKeyDir, "ubuntu-data.sealed-key"),
-		TPMPolicyAuthKeyFile: filepath.Join(InstallHostFDESaveDir, "tpm-policy-auth-key"),
-		TPMLockoutAuthFile:   filepath.Join(InstallHostFDESaveDir, "tpm-lockout-auth"),
+
+	// the boot chains we seal the fallback object to
+	rpbc := toPredictableBootChains(recoveryBootChains)
+
+	authKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```

Comment:
I was a bit confused initially where this authKey is saved. Maybe a small comment that the "seal*ObjectKeys()" will save the key to disk?